packages/api/lambdas/sf-event-sqs-to-db-records/write-execution.js

```diff
@@ -1,69 +1,157 @@
-const log = require('@cumulus/common/log');
+const isNil = require('lodash/isNil');
+
 const {
   tableNames,
 } = require('@cumulus/db');
 const {
   getMessageExecutionArn,
+  getExecutionUrlFromArn,
   getMessageCumulusVersion,
+  getMessageExecutionOriginalPayload,
+  getMessageExecutionFinalPayload,
 } = require('@cumulus/message/Executions');
 const {
   getMetaStatus,
+  getMessageWorkflowTasks,
+  getMessageWorkflowName,
+  getMessageWorkflowStartTime,
+  getMessageWorkflowStopTime,
 } = require('@cumulus/message/workflows');
 
+const { parseException } = require('../../lib/utils');
 const Execution = require('../../models/executions');
 
-const {
-  isPostRDSDeploymentExecution,
-  hasNoAsyncOpOrExists,
-  hasNoParentExecutionOrExists,
-} = require('./utils');
+const shouldWriteExecutionToPostgres = ({
+  messageCollectionNameVersion,
+  collectionCumulusId,
+  messageAsyncOperationId,
+  asyncOperationCumulusId,
+  messageParentExecutionArn,
+  parentExecutionCumulusId,
+}) => {
+  const noMessageCollectionOrExistsInPostgres = isNil(messageCollectionNameVersion)
```

Comment:
@Jkovarik (or whoever) - does this logic make sense? it's basically intending to say, "either there **is no** collection on the message or there **is** a collection in Postgres". Something about using the `||` makes me worry it's not safe, but there are unit tests here

Reply:
isNil(A) || !isNil(B) 

Works.   Something like "messageAsyncReferencesCorrect" for the variables might work better, though.